File: tests/conformance/fixtures/typescript/23-task-type.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('build').run('echo build').taskType('build');
p.task('test').run('echo test').taskType('test').after('build');
p.task('lint').run('echo lint').taskType('lint').after('test');
p.task('format').run('echo format').taskType('format').after('lint');
p.task('scan').run('echo scan').taskType('scan').after('format');
p.task('package').run('echo package').taskType('package').after('scan');
p.task('publish').run('echo publish').taskType('publish').after('package');
p.task('deploy').run('echo deploy').taskType('deploy').after('publish');
p.task('migrate').run('echo migrate').taskType('migrate').after('deploy');
p.task('generate').run('echo generate').taskType('generate').after('migrate');
p.task('verify').run('echo verify').taskType('verify').after('generate');
p.task('cleanup').run('echo cleanup').taskType('cleanup').after('verify');

p.emit();

File: tests/conformance/fixtures/elixir/23-task-type.exs
use Sykli

pipeline do
  task "build" do
    run "echo build"
    task_type :build
  end

  task "test" do
    run "echo test"
    task_type :test
    after_ ["build"]
  end

  task "lint" do
    run "echo lint"
    task_type :lint
    after_ ["test"]
  end

  task "format" do
    run "echo format"
    task_type :format
    after_ ["lint"]
  end

  task "scan" do
    run "echo scan"
    task_type :scan
    after_ ["format"]
  end

  task "package" do
    run "echo package"
    task_type :package
    after_ ["scan"]
  end

  task "publish" do
    run "echo publish"
    task_type :publish
    after_ ["package"]
  end

  task "deploy" do
    run "echo deploy"
    task_type :deploy
    after_ ["publish"]
  end

  task "migrate" do
    run "echo migrate"
    task_type :migrate
    after_ ["deploy"]
  end

  task "generate" do
    run "echo generate"
    task_type :generate
    after_ ["migrate"]
  end

  task "verify" do
    run "echo verify"
    task_type :verify
    after_ ["generate"]
  end

  task "cleanup" do
    run "echo cleanup"
    task_type :cleanup
    after_ ["verify"]
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
